### PR TITLE
Decisions

### PR DESCRIPTION
**Pull request by kodebach**
These are the mostly final decisions from #4223, which should be discussed at the next meeting.

### CODE REVIEW

**Comment by kodebach** [comment]
I will create separate PRs for the other less finalized decisions

File: doc/decisions/library_directory_structure.md
Comment (kodebach) [inline doc/decisions/library_directory_structure.md:52]: Previous discussion: https://github.com/ElektraInitiative/libelektra/pull/4223#discussion_r781492861

File: doc/decisions/library_directory_structure.md
Comment (kodebach) [inline doc/decisions/library_directory_structure.md:49]: @markus2330 @lawli3t I have now updated this document with the results from https://github.com/ElektraInitiative/libelektra/pull/4223#discussion_r809951795.

File: doc/decisions/private_api.md
Comment (kodebach) [inline doc/decisions/private_api.md:42]:
```suggestion
```

**Comment by kodebach** [comment]
@markus2330 I think this is ready to merge, when the build jobs finish

**Comment by markus2330** [comment]
There are a few minor things, will do the review later...

File: doc/decisions/library_directory_structure.md
Comment (markus2330) [inline doc/decisions/library_directory_structure.md:31]: I would only give -c prefixes if an alternative implementation exists. We plan to remove the C++ tools lib.

```suggestion
```

File: doc/decisions/library_directory_structure.md
Comment (markus2330) [inline doc/decisions/library_directory_structure.md:61]: I don't think we should introduce git submodules.

File: doc/decisions/private_api.md
Comment (markus2330) [inline doc/decisions/private_api.md:17]: shortly describe what you consider as "current situation" (to give contrast to your proposed solution).

File: doc/decisions/private_api.md
Comment (markus2330) [inline doc/decisions/private_api.md:24]: Is this already implemented in new-backend. We probably need an issue to track this.

File: doc/decisions/elektra_prefix.md
Comment (kodebach) [reply doc/decisions/elektra_prefix.md]: We remove this exact assumption in the meeting yesterday...

File: doc/decisions/library_directory_structure.md
Comment (kodebach) [reply doc/decisions/library_directory_structure.md]: > It would be amazing if you can start this work.

That is a hard no from me. I have far too much work to do already.

File: doc/decisions/library_directory_structure.md
Comment (kodebach) [reply doc/decisions/library_directory_structure.md:54]: It may be worded badly, but I don't think it's a big task. It's mostly about moving stuff from `tests/ctest` into the appropriate `src/libs/[lib]` folder. Then rest just needs to be moved from `tests` to `src/tests`. For Java/Rust the tests are already set up as described here.

File: doc/decisions/private_api.md
Comment (kodebach) [reply doc/decisions/private_api.md:17]: The "current situation" is hard to describe, would this be good enough:

```suggestion
- keep current situation: Theoretically there is `kdbprivate.h` and everything there is supposed to be private, but lots of code still uses it when it shouldn't. `kdb.h` is also the only header that is definitely public. All other headers are anybody's guess.
```

File: doc/decisions/private_api.md
Comment (kodebach) [reply doc/decisions/private_api.md:21]: How about:

```suggestion
- Also allow other libraries (e.g. a new `libelektra-operations`) to access to non-public API.
  Such libraries need to have a good reason (e.g. performance, impossible otherwise, etc.) why they access non-public API and they need to be kept up-to-date.
  If a library cannot provide a stable API on top of the unstable non-public API, it clearly needs to state which APIs are not stable.
```

File: doc/decisions/private_api.md
Comment (kodebach) [reply doc/decisions/private_api.md:24]: No it is not implemented. But we removed the extern consts for the version yesterday, so I'd just remove this from the decision too.

```suggestion
```

File: src/libs/elektra/README.md
Comment (kodebach) [reply src/libs/elektra/README.md:5]: Same as above

```suggestion
_Outdated_
```

File: doc/decisions/elektra_prefix.md
Comment (markus2330) [reply doc/decisions/elektra_prefix.md]: As said at another comment: these were old reviews that seemingly never were submitted.

File: doc/decisions/library_directory_structure.md
Comment (markus2330) [reply doc/decisions/library_directory_structure.md]: I'll make a new bachelor thesis topic about that.

Before that: lets keep the status quo. Then it is at least quite simple to run the existing benchmarks manually. (Moving to `src`, however, probably makes sense if you move the `tests`.)

File: doc/decisions/private_api.md
Comment (markus2330) [reply doc/decisions/private_api.md:21]: :+1:

File: src/libs/README.md
Comment (markus2330) [reply src/libs/README.md:29]: Let's discuss this in the meeting later.

**Comment by markus2330** [comment]
I would suggest that we concentrate on this PR getting merged. Please rework with the latest review.

**Comment by kodebach** [comment]
@markus2330 I have now incorporated the suggestions from the latest review. I haven't rebased yet. I'll wait until after the release to do that. Please check, if there is anything that still needs to change.

**Review by markus2330 — APPROVED** [review]
Looks good to me, please rebase.

**Pull request merged**
